src/Microsoft.ML.StaticPipe/LpNormalizerStaticExtensions.cs

```diff
@@ -47,16 +47,16 @@ public Reconciler(LpNormalizingEstimatorBase.NormFunction normKind, bool ensureZ
                 foreach (var outCol in toOutput)
                     pairs.Add((outputNames[outCol], inputNames[((OutPipelineColumn)outCol).Input]));
 
-                return new LpNormalizingEstimator(env, pairs.ToArray(), _norm, _ensureZeroMean);
+                return new LpNormNormalizingEstimator(env, pairs.ToArray(), _norm, _ensureZeroMean);
             }
         }
 
         /// <include file='../Microsoft.ML.Transforms/doc.xml' path='doc/members/member[@name="LpNormalize"]/*'/>
         /// <param name="input">The column to apply to.</param>
-        /// <param name="normKind">Type of norm to use to normalize each sample.</param>
-        /// <param name="subMean">Subtract mean from each value before normalizing.</param>
-        public static Vector<float> LpNormalize(this Vector<float> input,
-            LpNormalizingEstimatorBase.NormFunction normKind = LpNormalizingEstimatorBase.Defaults.Norm,
-            bool subMean = LpNormalizingEstimatorBase.Defaults.LpEnsureZeroMean) => new OutPipelineColumn(input, normKind, subMean);
+        /// <param name="norm">Type of norm to use to normalize each sample.</param>
+        /// <param name="ensureZeroMean">Subtract mean from each value before normalizing.</param>
+        public static Vector<float> LpNormNormalize(this Vector<float> input,
```

Comment:
>LpNormNormalize [](start = 36, length = 15)

is it too bad to keep it as is? NormNormalizing feels like a tongue twist :)

Reply:
That was suggested by Rogan for alignment with other normalizations. Let me try NormalizeLpNorm.

---
In reply to: [265323674](https://github.com/dotnet/machinelearning/pull/2888#discussion_r265323674) [](ancestors = 265323674)

Reply:
i like that better, personally

---
In reply to: [265327419](https://github.com/dotnet/machinelearning/pull/2888#discussion_r265327419) [](ancestors = 265327419,265323674)

Reply:
`LpNormalizer` normalizes *rows* and not *columns*, so maybe we should just call it `NormalizeRows`.

---
In reply to: [265328368](https://github.com/dotnet/machinelearning/pull/2888#discussion_r265328368) [](ancestors = 265328368,265327419,265323674)

Reply:
There are several normalizers around. We can't drop things like `LpNorm` and `GlobalContrast`.

---
In reply to: [265340604](https://github.com/dotnet/machinelearning/pull/2888#discussion_r265340604) [](ancestors = 265340604,265328368,265327419,265323674)